book/src/fork-checkpoints.md

```diff
@@ -0,0 +1,68 @@
+# Fork Checkpoints
+
+This design describes the software architecture for fork check pointing.  It addresses the following challenges:
+
+* Forks are created at every block.
+* Forks can be based on any previously produced block.
+* Forks are eventually fully committed such that rollback is impossible.
+* Unreachable forks need to be pruned.
+
+## Architecture
+
+The basic design idea is to maintain a DAG of forks.  Each fork points back to a single ancestor.  The DAG is initialized with a root.  Each subsequent fork must descend from the root.  
+
+## Active Chains
+
+An *active chain* is a direct list of connected forks that descend from the current root to a specific fork without any descendants.
```

Comment:
Can this acknowledge the `Blocktree` terminology?  Looks like "active chain" means the same thing as a path from a Blocktree leaf to the root.  Also, why introduce the term "chain" here? Why not an "active fork"?  Use of the word "active" is nicely consistent with GitHub's notion of a "active branch".